### PR TITLE
feat: Target net6.0

### PR DESCRIPTION
**Pull request by Morilli**
As mentioned in #132, this should be a simple upgrade without compromising support for any modern .NET versions.

I've also included a commit to use the new `ArgumentNullException.ThrowIfNull` which can be reverted if unwanted.

### CODE REVIEW

**Review by Crauzer — CHANGES_REQUESTED** [review]
I think we should keep using `Guard` instead of `ArgumentNullException.ThrowIfNull` because it will look a bit weird when you start mixing the two together.

**Comment by Morilli** [comment]
> I think we should keep using `Guard` instead of `ArgumentNullException.ThrowIfNull` because it will look a bit weird when you start mixing the two together.

Fair enough. I just assumed the built-in function may be advantageous to an external method, but I'm really indifferent about it. The static `ArgumentNullException` function does have the advantage of not generating this warning for me though:
image

File: src/LeagueToolkit/Meta/MetaEmbedded.cs [hunk 1]
```diff
@@ -9,17 +9,17 @@ public T Value
             get => this._value;
             set
             {
-                ArgumentNullException.ThrowIfNull(value, nameof(value));
-                this._value = value;
+                if (value is null) throw new ArgumentNullException(nameof(value));
```
Comment (Crauzer) [inline src/LeagueToolkit/Meta/MetaEmbedded.cs:12]: Can change to use `Guard` API

File: src/LeagueToolkit/Meta/MetaEmbedded.cs [hunk 2]
```diff
@@ -9,17 +9,17 @@ public T Value
...
             }
         }
 
         private T _value;
 
         public MetaEmbedded(T value)
         {
-            ArgumentNullException.ThrowIfNull(value, nameof(value));
-            this._value = value;
+            if (value is null) throw new ArgumentNullException(nameof(value));
```
Comment (Crauzer) [inline src/LeagueToolkit/Meta/MetaEmbedded.cs:21]: Can change to use `Guard` API

File: src/LeagueToolkit/Meta/MetaEmbedded.cs (hunk 1 above)
Comment (Morilli) [reply src/LeagueToolkit/Meta/MetaEmbedded.cs:12]: At this point I could also change all occurences of the pattern `if (x is null) throw new ArgumentNullException()` to use the Guard syntax if that's fine? Not sure why I even included this specific one in the first place.

**Comment by Crauzer** [comment]
> The static `ArgumentNullException` function does have the advantage of not generating this warning for me though: image

https://www.jetbrains.com/help/rider/PossibleMultipleEnumeration.html#false-positives

**Review by Crauzer — APPROVED** [review]

**Pull request merged**